### DISCUSSION TITLE
problem- Gramine with Node20 version (with intel-sgx)

### DISCUSSION THREAD

**Discussion by ROF13ThFloor** [gramineproject/gramine#1574] [Q&A]
Hi all, 

I'm trying to use Gramine to run the simple Nodejs application using node V20.0.0 inside the SGX enclave 

when I ran the application I got this error : 

![Screenshot from 2023-09-27 15-26-59](https://github.com/gramineproject/gramine/assets/34896746/6d63d14f-bb37-422d-b333-d895c2e0764a)

**Note: there is no problem with lower versions like 19.9.0**
Program is very simple 
`console.log("hello world")`

**Note: Gramine-direct has the same output**

My question is, does Gramine support node-20 and higher version?

**Comment by dimakuv** [comment]
Looks like Gramine has some problem with Node.js v20. I don't think anyone in the Gramine team tested this version.

@ROF13ThFloor Could we ask you to enable the Gramine trace log (`loader.log_level = "trace"` in the manifest template) and re-run your application, and attach the resulting log here.

**Reply by Villain88** [reply]
@ROF13ThFloor were you able to solve the problem?
We have the same issue

**Reply by dimakuv** [reply]
@Villain88 This was supposed to be fixed with the latest Gramine (v1.6.1), which we released a couple days ago. Alternatively, you can use the `master` branch of Gramine.

See https://github.com/gramineproject/gramine/pull/1692.

Unfortunately I forgot to mention this fact in this discussion.
